web/src/settings_users.ts

```diff
@@ -0,0 +1,741 @@
+            );
+            loading.destroy_indicator($("#admin_page_deactivated_users_loading_indicator"));
+            $("#admin_deactivated_users_table").show();
+        },
+        handle_events() {
+            const $tbody = $("#admin_deactivated_users_table").expectOne();
+
+            handle_filter_change($tbody, section.deactivated);
+            handle_deactivation($tbody);
+            handle_reactivation($tbody);
+            handle_edit_form($tbody);
+        },
+    },
+    bots: {
+        create_table() {
+            loading.make_indicator($("#admin_page_bots_loading_indicator"), {
+                text: $t({defaultMessage: "Loading…"}),
+            });
+            const $bots_table = $("#admin_bots_table");
+            $bots_table.hide();
+            const bot_user_ids = people.get_bot_ids();
+
+            bot_list_widget = ListWidget.create($bots_table, bot_user_ids, {
+                name: "admin_bot_list",
+                get_item: bot_info,
+                modifier_html: render_admin_user_list,
+                html_selector(item) {
+                    const user_id = item?.user_id?.toString() ?? "";
+                    return $(`tr[data-user-id='${CSS.escape(user_id)}']`);
+                },
+                filter: {
+                    $element: $bots_table
+                        .closest(".settings-section")
+                        .find<HTMLInputElement>(".search"),
+                    predicate(item, value) {
+                        if (!item) {
+                            return false;
+                        }
+                        return (
+                            item.full_name.toLowerCase().includes(value) ||
+                            (item.display_email?.toLowerCase().includes(value) ?? false)
+                        );
+                    },
+                    onupdate: reset_scrollbar($bots_table),
+                },
+                $parent_container: $("#admin-bot-list").expectOne(),
+                init_sort: "full_name_alphabetic",
+                sort_fields: {
+                    email: sort_bot_email,
+                    bot_owner: sort_bot_owner,
+                    role: sort_bot_role,
+                    ...ListWidget.generic_sort_functions("alphabetic", ["full_name", "bot_type"]),
+                },
+                $simplebar_container: $("#admin-bot-list .progressive-table-wrapper"),
+            });
+
+            loading.destroy_indicator($("#admin_page_bots_loading_indicator"));
+            $bots_table.show();
+        },
+        handle_events() {
+            const $tbody = $("#admin_bots_table").expectOne();
+
+            handle_bot_deactivation($tbody);
+            handle_reactivation($tbody);
+            handle_edit_form($tbody);
+        },
+    },
+};
+
+function sort_bot_role(a: BotInfo | undefined, b: BotInfo | undefined): number {
+    return user_sort.compare_a_b(a?.role ?? 0, b?.role ?? 0);
+}
```

Comment:
I made this to skip calling `user.sort_role` since that'll introduce an import cycle if BotInfo is imported there.

Reply:
You can use [`import type`](https://www.typescriptlang.org/docs/handbook/modules/reference.html#type-only-imports-and-exports) to avoid creating a runtime import cycle.